rtrlib/rtr/packets.c

```diff
@@ -366,7 +379,19 @@ static void rtr_pdu_convert_footer_byte_order(void *pdu, const enum target_byte_
 		((struct pdu_router_key *)pdu)->asn =
 			lrtr_convert_long(target_byte_order, ((struct pdu_router_key *)pdu)->asn);
 		break;
-
+            
+	case ASPA:
+		((struct pdu_aspa *)pdu)->provider_count =
+			lrtr_convert_short(target_byte_order, ((struct pdu_aspa *)pdu)->provider_count);
+		((struct pdu_aspa *)pdu)->customer_asn =
+			lrtr_convert_long(target_byte_order, ((struct pdu_aspa *)pdu)->customer_asn);
```

Comment:
why is this a long ? AS numbers should be `uin32_t`

Reply:
because of `uint32_t htonl(uint32_t hostlong)`, which gets called by `lrtr_convert_long

Reply:
that's very weird and irritating but okay